src/KnowbaseItem.php

```diff
@@ -2333,4 +2415,70 @@ public static function getIcon()
     {
         return "ti ti-lifebuoy";
     }
+
+    public static function getAdditionalSearchCriteria($params)
+    {
+        $kbi = new self();
+        if (!$kbi->canView()) {
+            $params['criteria'][] = [
+                'link'          => "AND",
+                'field'         => '8',
+                'searchtype'    => "equals",
+                'virtual'       => true,
+                'value'         => 2,
+            ];
+        } elseif (!Session::haveRight('knowbase', READ)) {
+            $params['criteria'][] = [
+                'link'          => "AND",
+                'field'         => '8',
+                'searchtype'    => "equals",
+                'virtual'       => true,
+                'value'         => 1,
+            ];
+        }
+
+        $unpublished = [
+            '0' => [
+                'link'          => "AND",
+                'field'         => '80',
+                'searchtype'    => "notunder",
+                'virtual'       => true,
+                'value'         => 0,
+            ],
+            '1' => [
+                'link'          => "AND",
+                'field'         => '81',
+                'searchtype'    => "equals",
+                'virtual'       => true,
+                'value'         => null,
```

Comment:
These `null` values are not working. Indeed, if I create a KB entry that has a profile as a target, I will not see it into "Published" results.